src/main/java/com/exasol/releasedroid/usecases/release/ReleaseInteractor.java

```diff
@@ -69,33 +70,72 @@ private void logResults(final Report releaseReport) {
 
     private Report makeRelease(final Repository repository, final List<PlatformName> platforms) {
         final var report = Report.releaseReport();
+        final Set<PlatformName> released = getAlreadyReleasedPlatforms(repository.getName(), repository.getVersion());
+        if (unreleasedPlatformsPresent(platforms, released)) {
+            report.merge(release(repository, platforms, released));
+        } else {
+            LOGGER.info(() -> "Nothing to release. The release has been already performed on all mentioned platforms.");
+        }
+        return report;
+    }
+
+    private Set<PlatformName> getAlreadyReleasedPlatforms(final String repositoryName, final String releaseVersion) {
+        return this.releaseState.getProgress(repositoryName, releaseVersion);
+    }
+
+    private boolean unreleasedPlatformsPresent(final List<PlatformName> platforms, final Set<PlatformName> released) {
+        return !released.containsAll(platforms);
+    }
+
+    private Report release(final Repository repository, final List<PlatformName> platforms,
+            final Set<PlatformName> released) {
         prepareRepositoryForRelease(repository);
-        report.merge(releaseOnPlatforms(repository, platforms));
-        cleanRepositoryAfterRelease(repository, report);
+        final Report report = releaseOnPlatforms(repository, platforms, released);
+        cleanRepositoryAfterRelease(repository, report.hasFailures());
         return report;
     }
 
-    private Report releaseOnPlatforms(final Repository repository, final List<PlatformName> platforms) {
+    private Report releaseOnPlatforms(final Repository repository, final List<PlatformName> platforms,
+            final Set<PlatformName> released) {
         final Report report = Report.releaseReport();
         for (final PlatformName platformName : platforms) {
-            LOGGER.info(() -> "Releasing on " + platformName + " platform.");
-            try {
-                this.getReleaseMaker(platformName).makeRelease(repository);
-                report.addResult(ReleaseResult.successfulRelease(platformName));
-            } catch (final ReleaseException exception) {
-                report.addResult(ReleaseResult.failedRelease(platformName, ExceptionUtils.getStackTrace(exception)));
-                break;
+            if (released.contains(platformName)) {
```

Comment:
would be nice to move this outside this method to be consistent and match the semantic of the method with its name